src/Testcontainers.Nats/NatsBuilder.cs

```diff
@@ -0,0 +1,108 @@
+namespace Testcontainers.Nats;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class NatsBuilder : ContainerBuilder<NatsBuilder, NatsContainer, NatsConfiguration>
+{
+    public const string NatsImage = "nats:2.9";
+
+    public const ushort ClientPort = 4222;
+    public const ushort RoutingPort = 6222;
+    public const ushort MonitoringPort = 8222;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NatsBuilder" /> class.
+    /// </summary>
+    public NatsBuilder()
+        : this(new NatsConfiguration())
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NatsBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private NatsBuilder(NatsConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override NatsConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Sets the Nats Server password.
+    /// </summary>
+    /// <param name="password">The Nats Server password.</param>
+    /// <returns>A configured instance of <see cref="NatsBuilder" />.</returns>
+    public NatsBuilder WithPassword(string password)
+    {
+        return Merge(DockerResourceConfiguration, new NatsConfiguration(password: password))
+            .WithCommand("-pass", password);
+    }
+
+    /// <summary>
+    /// Sets the Nats Server username.
+    /// </summary>
+    /// <param name="username">The Nats Server username.</param>
+    /// <returns>A configured instance of <see cref="NatsBuilder" />.</returns>
+    public NatsBuilder WithUsername(string username)
+    {
+        return Merge(DockerResourceConfiguration, new NatsConfiguration(username: username))
+            .WithCommand("-user", username);
```

Comment:
```suggestion
            .WithCommand("--user", username);
```

According to the documentation, it is using double dashes.